### ISSUE TITLE
Using font family multiple times results in default font

### ISSUE THREAD

**Issue by saschagehlich** [Automattic/node-canvas#977]
## Issue or Feature
When registering multiple font files with the same font family, but different styles (like in `examples/font.js`), the default font is used instead of the given font files. When using different font family names for every font file, it works.

Right now, `examples/font.js` creates this image:

![font2](https://user-images.githubusercontent.com/119684/29859100-da6b0ac4-8d60-11e7-8280-b70cb8b641de.png)

As you can see, the `Pfennig` font is not used at all.

Changing the font definitions like this (and changing the `font` property on the rendering context accordingly):

```js
Canvas.registerFont(fontFile('Pfennig.ttf'), {family: 'Pfennig'})
Canvas.registerFont(fontFile('PfennigBold.ttf'), {family: 'PfennigBold', weight: 'bold'})
Canvas.registerFont(fontFile('PfennigItalic.ttf'), {family: 'PfennigItalic', style: 'italic'})
Canvas.registerFont(fontFile('PfennigBoldItalic.ttf'), {family: 'PfennigBoldItalic', weight: 'bold', style: 'italic'})
```

This results in:

![font](https://user-images.githubusercontent.com/119684/29859081-c6c651cc-8d60-11e7-91ba-203c612cd19f.png)

## Your Environment
* Version of node-canvas (e.g. 1.4.0): 2.0.0-alpha.3
* Environment (e.g. node 4.2.0 on Mac OS X 10.8): node 7.9.0 on macOS 10.12.6

**Comment by chearon** [comment]
Spent an hour trying to figure out why, and then I remembered that I already know why. 

The three consecutive `registerFont` calls basically add 3 `PangoFontDescription`s to an internal list held in node-canvas. When you set `ctx.font = "pfennigFont"` it finds 3 matches, and [joins the real font names with commas](https://github.com/Automattic/node-canvas/blob/master/src/Canvas.cc#L753) before it sends the font query to Pango. So what gets sent to the OS in this case would be `{family: "Pfennig, Pfennig, Pfennig"}`. 

That's a totally valid query, and it works on Linux and Windows. But [this bug in Pango](https://bugzilla.gnome.org/show_bug.cgi?id=762873) causes the bug you're seeing. See my comment on the original `Canvas.registerFont` PR [here](https://github.com/Automattic/node-canvas/pull/715#issuecomment-236395162). I submitted a patch to Pango [here](https://bugzilla.gnome.org/attachment.cgi?id=332369&action=diff) but they haven't said anything, I'll see if I can ping Behdad or someone else, but in the meantime I could probably update the code so that it doesn't send duplicate font names, which would also fix it.

**Comment by saschagehlich** [comment]
Thanks a lot for investigating @chearon!

**Comment by chearon** [comment]
No problem, thanks for reporting. I plan on making a PR soon. This should work out-of-the-box on macOS.

**Comment by pravdomil** [comment]
see 
https://gitlab.gnome.org/GNOME/pango/merge_requests/4

**Comment by LinusU** [comment]
Should be fixed in pango 1.42.3? Could someone confirm?

**Comment by chearon** [comment]
Yep, fixed in exactly that version.
